### PR TITLE
handling status change events and showing the new status in UI

### PR DESCRIPTION
**Pull request by llali**
handling status change events and showing the new status in UI and adding a test to verify it.

### CODE REVIEW

File: src/views/statusView.ts [hunk 1]
```diff
@@ -157,7 +157,28 @@ export default class StatusView implements vscode.Disposable {
...
+        bar.statusLanguageService.text = Constants.updatingIntelliSenseLabel;
+        this.showStatusBarItem(fileUri, bar.statusLanguageService);
+    }
+
+    public getlanguageServiceStatus(fileUri: string): string {
+        let bar = this.getStatusBar(fileUri);
+        return bar.statusLanguageService.text;
+    }
+
+    public languageServiceStatusChanged(fileUri: string, status: string): void {
```
Comment (coquagli) [inline src/views/statusView.ts]: I think having both `languageServiceUpdating` and `languageServiceStatusChanged` is a little redundant. They could probably be merged into `languageServiceStatusChanged`, adding one more case to the `switch`.

File: src/models/constants.ts [hunk 2]
```diff
@@ -231,6 +231,7 @@ export const macOpenSslHelpButton = 'Help';
 export const macOpenSslHelpLink = 'https://github.com/Microsoft/vscode-mssql/wiki/OpenSSL-Configuration';
 
 export const sqlToolsServiceName = 'SQLToolsService';
+export const serviceInitializingOuputChannelName = 'SQLToolsService Initializing';
```
Comment (kevcunnane) [inline src/models/constants.ts]: Typo in name (ouput -> output) and suggest changing the text:
`export const serviceInitializingOutputChannelName = 'SqlToolsService Initialization';`

File: test/serviceClient.test.ts [hunk 3]
```diff
@@ -71,10 +75,31 @@ suite('Service Client tests', () => {
...
+            let fixture: IFixture = {
+                installedServerPath: 'already installed service',
+                downloadedServerPath: undefined,
+                platformInfo: new PlatformInformation('win32', 'x86_64', undefined)
+            };
+            const testFile = 'file:///my/test/file.sql';
+            const status = 'new status';
+
+            setupMocks(fixture);
+            let serviceClinet = new SqlToolsServiceClient(testServiceProvider.object, logger, testStatusView.object);
```
Comment (kevcunnane) [inline test/serviceClient.test.ts]: typo: serviceClient

File: src/models/constants.ts [hunk 4]
```diff
@@ -239,3 +240,9 @@ export const gettingStartedGuideLink = 'https://aka.ms/mssql-getting-started';
 export const sqlToolsServiceCrashMessage = 'SQL Tools Service component could not start.';
 export const sqlToolsServiceCrashButton = 'View Known Issues';
 export const sqlToolsServiceCrashLink = 'https://github.com/Microsoft/vscode-mssql/wiki/SqlToolsService-Known-Issues';
+
+export const gettingdefinitionMessage = 'Getting definition ...';
```
Comment (kevcunnane) [inline src/models/constants.ts]: Camelcase: `gettingDefinitionMessage` with capital D

**Pull request merged**